ImperatorToCK3Tests/MapperTests/RegionMapper/CK3RegionTests.cpp

```diff
@@ -0,0 +1,99 @@
+#include "../../ImperatorToCK3/Source/Mappers/RegionMapper/CK3Region.h"
+#include "gtest/gtest.h"
+#include <sstream>
+
+TEST(Mappers_RegionTests, blankRegionLoadsWithNoRegionsAndNoDuchies)
+{
+	std::stringstream input;
+	const mappers::CK3Region region(input);
+
+	ASSERT_TRUE(region.getRegions().empty());
+	ASSERT_TRUE(region.getDuchies().empty());
+}
+
+TEST(Mappers_RegionTests, areaCanBeLoaded)
+{
+	std::stringstream input;
+	input << "duchies = { d_ivrea } \n";
+	const mappers::CK3Region region(input);
+
+	ASSERT_FALSE(region.getDuchies().empty());
+	ASSERT_EQ("d_ivrea", region.getDuchies().find("d_ivrea")->first);
+}
+
+TEST(Mappers_RegionTests, regionCanBeLoaded)
+{
+	std::stringstream input;
+	input << "regions = { sicily_region } \n";
+	const mappers::CK3Region region(input);
+
+	ASSERT_FALSE(region.getRegions().empty());
+	ASSERT_EQ("sicily_region", region.getRegions().find("sicily_region")->first);
+}
+
+TEST(Mappers_RegionTests, multipleDuchiesCanBeLoaded)
+{
+	std::stringstream input;
+	input << "duchies = { d_ivrea d_athens d_oppo } \n";
+	const mappers::CK3Region region(input);
+
+	ASSERT_EQ(3, region.getDuchies().size());
+}
+
+TEST(Mappers_RegionTests, multipleRegionsCanBeLoaded)
+{
+	std::stringstream input;
+	input << "regions = { sicily_region island_region new_region } \n";
+	const mappers::CK3Region region(input);
+
+	ASSERT_EQ(3, region.getRegions().size());
+}
+
+TEST(Mappers_RegionTests, regionCanBeLinkedToDuchy)
+{
+	std::stringstream input;
+	input << "duchies = { d_ivrea d_athens d_oppo } \n";
+	mappers::CK3Region region(input);
+
+	std::stringstream input2;
+	input2 << "{ c_athens = { b_athens = { province = 79 } b_newbarony = { province = 56 } } } \n";
+	auto duchy2 = std::make_shared<CK3::Title>("d_athens");
+	duchy2->loadTitles(input2);
+
+	ASSERT_FALSE(region.getDuchies().find("d_athens")->second); // nullptr before linking
```

Comment:
Test for equality to nullptr instead of testing for false

Reply:
Not convinced. while testing for nullptr makes sense is indeed correct with regards to what's happening behind the scenes, that's not actual use case within the converter. We don't do if(thing== nullptr), we do if(thing), effectively testing true or false.
I believe we whould test in the same manner as we use the components, no?

Reply:
I'm kinda on the fence. It's true that we use `if(thing)`, but I think it also makes sense to check in a test if the value is actually nullptr instead of 0.